library/gcm.c

```diff
@@ -412,14 +412,14 @@ int mbedtls_gcm_starts(mbedtls_gcm_context *ctx,
         while (iv_len > 0) {
             use_len = (iv_len < 16) ? iv_len : 16;
 
-#if defined(MBEDTLS_COMPILER_IS_GCC)
+#if defined(MBEDTLS_COMPILER_IS_GCC) && (MBEDTLS_GCC_VERSION >= 70110)
```

Comment:
Why 70100 and not 70000? https://gcc.gnu.org/gcc-7/changes.html announces `Wstringop-overflow` in 7.0.

Reply:
I tested using Godbolt - this was the oldest version I could find for which the pragma definitely works.

Reply:
I'd rather leave the warning unfixed for users with gcc 7.0 using -O3, than risk introducing a regression.